mmengine/registry/registry.py

```diff
@@ -307,9 +307,16 @@ def infer_scope() -> str:
         # frame-0: `infer_scope` itself
         # frame-1: `__init__` of `Registry` which calls the `infer_scope`
         # frame-2: Where the `Registry(...)` is called
-        filename = inspect.getmodule(sys._getframe(2)).__name__  # type: ignore
-        split_filename = filename.split('.')
-        return split_filename[0]
+        try:
+            filename = inspect.getmodule(  # type: ignore
+                sys._getframe(2)).__name__
+            split_filename = filename.split('.')
+            scope = split_filename[0]
+        except AttributeError:
+            # use "None" to handle some cases which can not infer the scope
+            # like initializing Registry in interactive mode
+            scope = 'None'
```

Comment:
If the scope is None, will there be some problems when calling `build` or `get`? Especially when a none-scope node is added to a registry tree.

Reply:
The scope is a `'None'` string rather than `None` object.

Reply:
I set it as 'mmengine'